### ISSUE TITLE
[java] DoubleCheckedLocking is not detected if more than 1 assignment or more than 2 if statements

### ISSUE THREAD

**Issue by csrma** [pmd/pmd#5880]
**Affects PMD Version:** 7.15.0

**Rule:** DoubleCheckedLocking

Please provide the rule name and a link to the rule documentation: [DoubleCheckedLocking](https://docs.pmd-code.org/latest/pmd_rules_java_multithreading.html#doublecheckedlocking)

**Description:**
The [rule implementation](https://github.com/pmd/pmd/blob/39f7a60c2b2f88c03181fafc6179d01f1260414f/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/multithreading/DoubleCheckedLockingRule.java#L102) currently specifically checks for 2 if blocks (`isl.size() == 2`) and 1 assignment inside the second if block. However, it might be possible to have more than 1 assignments. It is a common pattern to add wrappers or other enhancement to an object after initializing (as exemplified below). In such cases, more than 1 assignments might be necessary.

**Code Sample demonstrating the issue:**

```java
class DoubleCheckedLockingBug {
    Object baz = null;

    Object bar() {
        if (baz == null) {
            synchronized(this) {
                if (baz == null) {
                    baz = new Object();
                    baz = enhanceObject(baz);
                }
            }
        }
        return baz;
    }

    private Object enhanceObject(Object obj) {
        // do special stuff
        return obj;
    }
}
```
Having more than 2 if-blocks in the `bar()` method will also result in PMD missing this use case. For example, the below code is not detected because of the additional `if (false)` statement.

```java
class DoubleCheckedLockingBug {
    Object baz = null;

    Object bar() {
        if (baz == null) {
            synchronized(this) {
                if (baz == null) {
                    baz = new Object();
                    baz = enhanceObject(baz);
                }
            }
        }
        if (false)
            System.out.println("This will never print.");
        return baz;
    }
}
```

**Expected outcome:**

PMD should report a violation at line 4, but doesn't. This is a false-negative.

**Running PMD through:** *[CLI]*
